solr/core/src/java/org/apache/solr/handler/admin/CoreAdminHandler.java

```diff
@@ -218,13 +217,16 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
       if (taskId == null) {
         callInfo.call();
       } else {
-        coreAdminAsyncTracker.submitAsyncTask(
-            taskObject,
-            action,
+        Callable<SolrQueryResponse> task =
             () -> {
               callInfo.call();
               return callInfo.rsp;
-            });
+            };
+
+        final CoreAdminAsyncTracker.TaskObject taskObject =
```

Comment:
Just want to mention that Java 11 "var" keyword would be useful here where there is a very long type, and we know the type clearly here anyway.  Or just inline this.